code/modules/events/pirates.dm

```diff
@@ -19,63 +15,40 @@
-			threat.possible_answers = list("We'll pay.","Tribute? Really? Go away.")
-		if(PIRATES_DUTCHMAN)
-			ship_name = "Flying Dutchman"
-			ship_template = /datum/map_template/shuttle/pirate/dutchman
-			threat.title = "Business proposition"
-			threat.content = "Ahoy! This be the [ship_name]. Cough up [payoff] credits or you'll walk the plank."
-			threat.possible_answers = list("We'll pay.","We will not be extorted.")
-	threat.answer_callback = CALLBACK(GLOBAL_PROC, .proc/pirates_answered, threat, payoff, ship_name, initial_send_time, response_max_time, ship_template)
-	addtimer(CALLBACK(GLOBAL_PROC, .proc/spawn_pirates, threat, ship_template, FALSE), response_max_time)
-	SScommunications.send_message(threat,unique = TRUE)
-
-/proc/pirates_answered(datum/comm_message/threat, payoff, ship_name, initial_send_time, response_max_time, ship_template)
-	if(world.time > initial_send_time + response_max_time)
-		priority_announce("Too late to beg for mercy!",sender_override = ship_name)
+	threat.answer_callback = CALLBACK(GLOBAL_PROC, .proc/pirates_answered, threat, chosen_gang, payoff, initial_send_time = world.time)
+	addtimer(CALLBACK(GLOBAL_PROC, .proc/spawn_pirates, threat, chosen_gang, FALSE), RESPONSE_MAX_TIME)
+	SScommunications.send_message(threat, unique = TRUE)
+
+/proc/pirates_answered(datum/comm_message/threat, payoff, datum/pirate_gang/chosen_gang, initial_send_time)
+	if(world.time > initial_send_time + RESPONSE_MAX_TIME)
+		priority_announce(chosen_gang.response_too_late ,sender_override = chosen_gang.ship_name)
 		return
 	if(threat && threat.answered == 1)
-		var/datum/bank_account/D = SSeconomy.get_dep_account(ACCOUNT_CAR)
-		if(D)
-			if(D.adjust_money(-payoff))
-				priority_announce("Thanks for the credits, landlubbers.",sender_override = ship_name)
-				return
+		var/datum/bank_account/plundered_account = SSeconomy.get_dep_account(ACCOUNT_CAR)
+		if(plundered_account)
+			if(plundered_account.adjust_money(-payoff))
+				priority_announce(chosen_gang.response_received, sender_override = chosen_gang.ship_name)
 			else
-				priority_announce("Trying to cheat us? You'll regret this!",sender_override = ship_name)
-				spawn_pirates(threat, ship_template, TRUE)
+				priority_announce(chosen_gang.response_not_enough, sender_override = chosen_gang.ship_name)
+				spawn_pirates(threat, chosen_gang, TRUE)
 
-/proc/spawn_pirates(datum/comm_message/threat, ship_template, skip_answer_check)
+/proc/spawn_pirates(datum/comm_message/threat, datum/pirate_gang/chosen_gang, skip_answer_check)
 	if(!skip_answer_check && threat?.answered == 1)
 		return
 
 	var/list/candidates = poll_ghost_candidates("Do you wish to be considered for pirate crew?", ROLE_TRAITOR)
 	shuffle_inplace(candidates)
 
-	var/datum/map_template/shuttle/pirate/ship = new ship_template
+	var/datum/map_template/shuttle/pirate/ship = new chosen_gang.ship_template
```

Comment:
I guess it's worth noting now, but...

The Mapping subsystem pre-loads all the shuttle templates (`SSmapping.shuttle_templates`). I'm pretty sure creating a new template is a waste, since you can just get the preloaded, instantiated datum from SSmapping.

Reply:
I'll test to see if this works